codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -786,337 +786,337 @@ let checkAndFetchForChain = (
-          //on the last batch and there are no items on the queue
-          dispatchAction(StartIndexingAfterPreRegister)
-        | {val: None} if state.chainManager->ChainManager.isFetchingAtHead =>
-          //pre registration is done, there are no items on the queue and we are fetching at head
-          //this case is only hit if we are indexing chains with no reorg threshold
-          dispatchAction(StartIndexingAfterPreRegister)
-        | _ => () //Nothing to process and pre registration is not done
+        switch await EventProcessing.getDynamicContractRegistrations(
+          ~latestProcessedBlocks,
+          ~eventBatch=batch,
+          ~checkContractIsRegistered,
+        ) {
+        | Ok(batchProcessed) => dispatchAction(DynamicContractPreRegisterProcessed(batchProcessed))
+        | Error(errHandler) => dispatchAction(ErrorExit(errHandler))
+        | exception exn =>
+          //All casese should be handled/caught before this with better user messaging.
+          //This is just a safety in case something unexpected happens
+          let errHandler =
+            exn->ErrorHandling.make(
+              ~msg="A top level unexpected error occurred during pre registration of dynamic contracts",
+            )
+          dispatchAction(ErrorExit(errHandler))
         }
+      | {isInReorgThreshold: true, val: None} =>
+        //pre registration is done, we've hit the multichain reorg threshold
+        //on the last batch and there are no items on the queue
+        dispatchAction(StartIndexingAfterPreRegister)
+      | {val: None} if state.chainManager->ChainManager.isFetchingAtHead =>
+        //pre registration is done, there are no items on the queue and we are fetching at head
+        //this case is only hit if we are indexing chains with no reorg threshold
+        dispatchAction(StartIndexingAfterPreRegister)
+      | _ => () //Nothing to process and pre registration is not done
+      }
+    }
+  | ProcessEventBatch =>
+    if !state.currentlyProcessingBatch && !isRollingBack(state) {
+      //Allows us to process events all the way up until we hit the reorg threshold
+      //across all chains before starting to capture entity history
+      let onlyBelowReorgThreshold = if state.config->Config.shouldRollbackOnReorg {
+        state.chainManager.isInReorgThreshold ? false : true
+      } else {
+        false
       }
-    | ProcessEventBatch =>
-      if !state.currentlyProcessingBatch && !isRollingBack(state) {
-        switch state.chainManager->ChainManager.createBatch(
-          ~maxBatchSize=state.maxBatchSize,
-          //Allows us to process events all the way up until we hit the reorg threshold
-          //across all chains before starting to capture entity history
-          ~onlyBelowReorgThreshold=state.chainManager.isInReorgThreshold ? false : true,
+      switch state.chainManager->ChainManager.createBatch(
+        ~maxBatchSize=state.maxBatchSize,
+        ~onlyBelowReorgThreshold,
```

Comment:
Please review with whitespace off (formatting changes)

This code is the only thing I changed. 

The bug was that it would never set state.chainManager.isInReorgThreshold to true when `rollback_on_reorg` is disabled. This way we skip that mode entirely.